src/utils.ts

```diff
@@ -90,22 +96,26 @@ export function loadScript(id: string, config: LoadScriptOptions): void {
     id,
     ...(config.queryParams ?? {})
   });
-  script.src = `https://www.googletagmanager.com/gtm.js?${queryString}`;
+
+  const source: string = config.source ?? 'https://www.googletagmanager.com/gtm.js';
+
+  script.src = `${source}?${queryString}`;
 
   const parentElement: HTMLElement = config.parentElement ?? doc.body;
+
   if (typeof parentElement?.appendChild !== 'function') {
     throw new Error('parentElement must be a DOM element');
   }
+
   parentElement.appendChild(script);
 }
 
 /**
  * Check if GTM script is in the document.
  *
- * @returns `true` if in the `document` is a `script` with `src` containing `googletagmanager.com/gtm.js`, otherwise `false`.
+ * @param source The URL of the script, if it differs from the default. Default: 'https://www.googletagmanager.com/gtm.js'.
+ * @returns `true` if in the `document` is a `script` with `src` containing "https://www.googletagmanager.com/gtm.js" (or `source` if specified), otherwise `false`.
```

Comment:
```suggestion
 * @returns `true` if in the `document` is a `script` with `src` containing `'https://www.googletagmanager.com/gtm.js'` (or `source` if specified), otherwise `false`.
```